doc/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+if (DOXYGEN_FOUND)
```

Comment:
With this commit, there will be a `doc` folder and a `docs` folder.
We will have to put everything in the `doc` folder, but that can be done later.

Reply:
@arnopo I discussed this with @danmilea and his perspective was, "The doc directory mirrors libmetal and it contains rules to build the docs. open-amp/docs on the other hand already contains some docs in mardown format. I wouldn't mix them."

Reply:
@arnopo I have removed the docs folder and placed everything in the doc folder.